### PR TITLE
Ensure addresspool is addressfamily agnostic

### PR DESCRIPTION
**Pull request by majst01**
## Description

None

### CODE REVIEW

**Review by iljarotar — APPROVED** [review]

File: pkg/controllers/loadbalancer/addresspool.go [hunk 1]
```diff
@@ -14,7 +15,7 @@ type AddressPool struct {
 	Name       string   `json:"name" yaml:"name"`
 	Protocol   string   `json:"protocol" yaml:"protocol"`
 	AutoAssign *bool    `json:"auto-assign" yaml:"auto-assign,omitempty"`
-	CIDRs      []string `json:"addresses,omitempty" yaml:"addresses,omitempty"` // It is assumed that only /32 addresses are used.
+	CIDRs      []string `json:"addresses,omitempty" yaml:"addresses,omitempty"` // It is assumed that only Host addresses (/32 for ipv4 or /128 for ipv6) are used.
```
Comment (robertvolkmann) [inline pkg/controllers/loadbalancer/addresspool.go:18]:
```suggestion
	CIDRs      []string `json:"addresses,omitempty" yaml:"addresses,omitempty"` // It is assumed that only host addresses (/32 for ipv4 or /128 for ipv6) are used.
```

File: pkg/controllers/loadbalancer/metallb.go [hunk 2]
```diff
@@ -58,7 +58,10 @@ func (cfg *MetalLBConfig) computeAddressPools(ips []*models.V1IPResponse, nws se
 			continue
 		}
 		net := *ip.Networkid
-		cfg.addIPToPool(net, *ip)
+		err := cfg.addIPToPool(net, *ip)
+		if err != nil {
+			return err
+		}
```
Comment (robertvolkmann) [inline pkg/controllers/loadbalancer/metallb.go:64]: Only the IP addresses before the first invalid one would be added.

**Review by robertvolkmann — CHANGES_REQUESTED** [review]

File: pkg/controllers/loadbalancer/addresspool.go (hunk 1 above)
Comment (majst01) [reply pkg/controllers/loadbalancer/addresspool.go]: fixed

File: pkg/controllers/loadbalancer/metallb.go (hunk 2 above)
Comment (majst01) [reply pkg/controllers/loadbalancer/metallb.go]: fixed

**Review by robertvolkmann — APPROVED** [review]

**Pull request merged**